### DISCUSSION TITLE
About the `Stop Charging When Sleeping` option

### DISCUSSION THREAD

**Discussion by honeySleepr** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#826] [Q&A] [answered]
Hi!
I have a question.

The description at the [apphousekitchen page](https://apphousekitchen.com/feature-explanation-stop-charging-when-sleeping/) says that if the `Stop Charging When Sleeping` is turned OFF and you let your mac sleep while charging, the battery will charge to 100%, regardless of the set limit.
> When your MacBook is sleeping, third-party apps cannot run and make any changes.

I also read a [related discussion](https://github.com/davidwernhart/AlDente-Charge-Limiter/discussions/605#discussioncomment-2250910) where @MatthiasKerbl answered
> .. When it is still charging to the charge limit and it goes into sleep, it will charge past the charge limit to 100%.

However, I checked last night that the battery was at 30% and still charging when I closed the lid, but in the morning the charging had stopped at 80%(the limit I set).
Has this feature been updated?

* Using Apple M1, Ventura 13.0, Aldente Pro 1.19.2

Thanks for this wonderful app 👍

**Comment by MatthiasKerbl** [comment, marked as answer]
Hi @honeySleepr,

Thank you for your question and I understand your confusion.
There is a lot of guessing in regard to this topic. So please don't take everything for 100% correct.

> When your MacBook is sleeping, third-party apps cannot run and make any changes.

In general, this is correct, however, since the introduction of the Apple Silicon Macs, this does not seem to be the case all the time anymore. Intel MacBooks had the option to activate the "Power Nap" feature in settings which would wake up the machine once in a while to check for emails and so on. On Apple Silicon Macs, this feature is completely gone and got replaced by a new, different version of it. This means, Apple Silicon Macs never completely sleep, and maybe third-party apps also sometimes get some CPU time which results in AlDente being able to stop charging even if your MacBook is theoretically asleep. However, this seems to be inconsistent and does not seem to work for everyone 100% of the time and not at all for Intel MacBooks. That's why we implemented the features "Stop charging when sleeping" and "Disable sleep until charge limit" to bring 100% consistent behavior to all of our users.

Hope that helps to understand this topic better. :)

Cheers,
Matthias

**Reply by honeySleepr** [reply]
Thank you, I understand now 😄
